### DISCUSSION TITLE
I encountered a question when mysql table have been using a datetime column

### DISCUSSION THREAD

**Discussion by yuer1727** [diesel-rs/diesel#2980] [Q&A]
Compiling fuso v0.1.0 (/Users/jiaming.rjm/work/rust_repository/fuso)
error[E0277]: the trait bound `chrono::naive::datetime::NaiveDateTime: diesel::Queryable<Datetime, Mysql>` is not satisfied
  --> src/main.rs:46:31
   |
46 |         .load::<WorkHolidays>(&connection)
   |          ----                 ^^^^^^^^^^^ the trait `diesel::Queryable<Datetime, Mysql>` is not implemented for `chrono::naive::datetime::NaiveDateTime`
   |          |
   |          required by a bound introduced by this call
   |
   = note: required because of the requirements on the impl of `diesel::Queryable<(Integer, Datetime, Integer, Integer, TinyInt, diesel::sql_types::Nullable<Text>, Text, Text, Datetime, Datetime), Mysql>` for `(i32, chrono::naive::datetime::NaiveDateTime, i32, i32, i8, Option<std::string::String>, std::string::String, std::string::String, chrono::naive::datetime::NaiveDateTime, chrono::naive::datetime::NaiveDateTime)`
   = note: 1 redundant requirement hidden
   = note: required because of the requirements on the impl of `diesel::Queryable<(Integer, Datetime, Integer, Integer, TinyInt, diesel::sql_types::Nullable<Text>, Text, Text, Datetime, Datetime), Mysql>` for `WorkHolidays`
   = note: required because of the requirements on the impl of `LoadQuery<_, WorkHolidays>` for `SelectStatement<fuso_diesel::schema::work_holidays::table, query_builder::select_clause::DefaultSelectClause, query_builder::distinct_clause::NoDistinctClause, query_builder::where_clause::NoWhereClause, query_builder::order_clause::NoOrderClause, query_builder::limit_clause::LimitClause<diesel::expression::bound::Bound<BigInt, i64>>>`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `fuso` due to previous error

**Comment by yuer1727** [comment]
i saw somebody meet the problem, but i don't found the solution.

**Comment by yuer1727** [comment]
**models.rs**
#[derive(Queryable, Debug)]
pub struct WorkHolidays {
    pub id: i32,
    pub day_time: chrono::NaiveDateTime,
    pub year: i32,
    pub month: i32,
    pub type_: i8,
    pub remark: Option<String>,
    pub last_operator: String,
    pub last_operator_id: String,
    pub create_time: chrono::NaiveDateTime,
    pub modify_time: chrono::NaiveDateTime,
}

**schema.rs**
table! {
    work_holidays (id) {
        id -> Integer,
        day_time -> Datetime,
        year -> Integer,
        month -> Integer,
        #[sql_name = "type"]
        type_ -> Tinyint,
        remark -> Nullable<Varchar>,
        last_operator -> Varchar,
        last_operator_id -> Varchar,
        create_time -> Datetime,
        modify_time -> Datetime,
    }
}

**Comment by weiznich** [comment]
First of all: Our bug tracker is not a question or support forum. It's meant to track actual bugs. Additionally for posts in our issue tracker it's required to follow the issue template (if not explicitly requested otherwise by one of the maintainers). As of this I've moved this to the discussion part.

Based on the information you provided it seems like you just missed to enable the `chrono` feature for diesel to enable the support for types coming from the `chrono` crate. It's hard to tell that for sure as your do not provide necessary information about the enabled feature flags in your post.

**Reply by yuer1727** [reply]
i got it!  Thanks very much . I don't found the docs describing about 'enable the chrono'.
